### PR TITLE
fix(private_locations): returning the right err

### PR DESCRIPTION
**Pull request by NSSPKrishna**
# Description

Please include a summary of the change and which issue is fixed (if relevant).

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [ ] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Step 1
- Step 2
- etc

### CODE REVIEW

**Review by mbazhlekova — APPROVED** [review]

File: newrelic/resource_newrelic_synthetics_private_location.go [hunk 1]
```diff
@@ -114,13 +115,15 @@ func resourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *schem
...
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
+		if err.Error() == "Argument \"guid\" has invalid value $guid." {
+			return diag.FromErr(fmt.Errorf("invalid GUID"))
 		}
-
 		return diag.FromErr(err)
 	}
+	if _, ok := err.(*errors.NotFound); ok {
```
Comment (kidk) [inline newrelic/resource_newrelic_synthetics_private_location.go:123]: Why did you move this code out of the err != nil if statement?

File: newrelic/resource_newrelic_synthetics_private_location.go [hunk 2]
```diff
@@ -114,13 +115,15 @@ func resourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *schem
 
 	resp, err := client.Entities.GetEntityWithContext(ctx, guid)
 	if err != nil {
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
+		if err.Error() == "Argument \"guid\" has invalid value $guid." {
+			return diag.FromErr(fmt.Errorf("invalid GUID"))
```
Comment (kidk) [inline newrelic/resource_newrelic_synthetics_private_location.go:119]: Can we add the value of the GUID that the user passed? There's been some complaints about this.

Comment (NSSPKrishna) [reply newrelic/resource_newrelic_synthetics_private_location.go:119]: Sure

File: newrelic/resource_newrelic_synthetics_private_location.go (hunk 1 above)
Comment (kidk) [reply newrelic/resource_newrelic_synthetics_private_location.go:123]: I would keep it for now. The goal is to get the errors.NotFound working again at some point.

**Review by kidk — APPROVED** [review]

File: newrelic/resource_newrelic_synthetics_private_location.go (hunk 1 above)
Comment (NSSPKrishna) [reply newrelic/resource_newrelic_synthetics_private_location.go]: done
